seqr/views/utils/dataset_utils.py

```diff
@@ -147,14 +147,23 @@ def _update_samples_for_dataset(
 def _get_elasticsearch_index_samples(elasticsearch_index):
     sample_field_suffix = '_num_alt'
 
-    index = elasticsearch_dsl.Index('{}*'.format(elasticsearch_index), using=get_es_client())
+    es_client = get_es_client()
+    index = elasticsearch_dsl.Index('{}*'.format(elasticsearch_index), using=es_client)
     try:
-        field_mapping = index.get_field_mapping(fields=['*{}'.format(sample_field_suffix)], doc_type=[VARIANT_DOC_TYPE])
+        field_mapping = index.get_field_mapping(fields=['*{}'.format(sample_field_suffix), 'join_field'], doc_type=[VARIANT_DOC_TYPE])
     except NotFoundError:
         raise Exception('Index "{}" not found'.format(elasticsearch_index))
     except TransportError as e:
         raise Exception(e.error)
 
+    #  Nested genotypes
+    if field_mapping.get(elasticsearch_index, {}).get('mappings', {}).get(VARIANT_DOC_TYPE, {}).get('join_field'):
+        s = elasticsearch_dsl.Search(using=es_client, index=elasticsearch_index)
+        s = s.params(size=0)
+        s.aggs.bucket('sample_ids', elasticsearch_dsl.A('terms', field='sample_id'))
+        response = s.execute()
+        return [agg['key'] for agg in response.aggregations.sample_ids.buckets]
```

Comment:
wouldn't you need to call elasticsearch_datastore._decode_name on this (and also in the non-nested version)?

Reply:
I forgot. But thinking about it we shouldn't need to encode/decode names for nested sample IDs anyways so I will take out the encoding here, rather than add the decoding below

Reply:
oh I misunderstood your comment, you meant we should be decoding the sample ids that aren't nested. i'll add that